app/controllers/application_controller.rb

```diff
@@ -3,12 +3,26 @@ class ApplicationController < ActionController::API
   private
 
   set_current_tenant_through_filter
+  before_action :set_current_headers
   before_action :set_the_current_tenant
+  after_action :remove_current_headers_and_tenant
+
+  def set_current_headers
+    ManageIQ::API::Common::Request.current = request
+  end
+
+  def remove_current_headers_and_tenant
+    ManageIQ::API::Common::Request.current = nil
+    ActsAsTenant.current_tenant = nil
+  end
 
   def set_the_current_tenant
     return unless Tenant.tenancy_enabled?
-
-    account_number = identity_account_number
+    begin
+      account_number = ManageIQ::API::Common::Request.current.user.tenant rescue nil
```

Comment:
you already rescue all how can you rescue later `HeaderIdentityError`?

Reply:
Yeah, that `rescue` should have been removed